### PR TITLE
[MIG] account_payment_partner: Migration to 13.0

### PR DESCRIPTION
**Pull request by rven**
Depends on:

- [x] account_payment_mode https://github.com/OCA/bank-payment/pull/636

### CODE REVIEW

**Comment by rven** [comment]
Hi @pedrobaeza,

How can I add the dependency on another PR?
Is this possible with oca_dependencies.txt. If yes, how should it be referenced?

**Comment by pedrobaeza** [comment]
I'm afraid it's not possible with oca_dependencies.txt. One technique is to add the commits of the other module here, and rebase after the merge. Another is to simply wait until the first module is merged, and then rebase. Last one is to modify temporary .travis.yml for download and inject the folder in the build path for having it available. I know some contributors have done it, but I don't have right now any example.

**Comment by pedrobaeza** [comment]
You can rebase now

File: account_payment_partner/models/account_move.py
Comment (astirpe) [inline account_payment_partner/models/account_move.py]: Method `_prepare_refund()` does not exist anymore in V13. You may want to use method `_reverse_moves()` instead.

File: account_payment_partner/models/account_move_line.py
Comment (astirpe) [inline account_payment_partner/models/account_move_line.py]: Why did you make it related?

**Review by astirpe — CHANGES_REQUESTED** [review]

File: account_payment_partner/models/account_move_line.py
Comment (rven) [reply account_payment_partner/models/account_move_line.py]: It's now a computed field

File: account_payment_partner/models/account_move.py
Comment (rven) [reply account_payment_partner/models/account_move.py]: Indeed, the function is renamed to _reverse_move_vals

File: account_payment_partner/models/account_payment_mode.py
Comment (astirpe) [inline account_payment_partner/models/account_payment_mode.py]: This constraint shouldn't be removed

File: account_payment_partner/models/account_move.py
Comment (astirpe) [inline account_payment_partner/models/account_move.py]: `readonly=True` is the default for related fields and can be removed here

File: account_payment_partner/models/account_move.py
Comment (astirpe) [inline account_payment_partner/models/account_move.py]: This decorator is wrong for `def _reverse_move_vals()`, must be removed.

File: account_payment_partner/models/account_move.py
Comment (astirpe) [inline account_payment_partner/models/account_move.py]: You can remove the parameters of the `super()` here and everywhere else in the code.

File: account_payment_partner/__manifest__.py
Comment (astirpe) [inline account_payment_partner/__manifest__.py]: You could rename this file to `account_move_view,xml`

File: account_payment_partner/models/account_move.py
Comment (astirpe) [inline account_payment_partner/models/account_move.py]: If you wrap the method name with quotes, like for example here `compute="_compute_payment_mode"`, you don't need to move the field definition below the compute method. This is valid in general: it's useful to limit the changes in the diff and ease the code review.

File: account_payment_partner/models/res_partner.py
Comment (astirpe) [inline account_payment_partner/models/res_partner.py]: Is this comment still needed?

File: account_payment_partner/models/res_partner.py
Comment (astirpe) [inline account_payment_partner/models/res_partner.py]: `string="Supplier Payment Mode",` could be removed here, as it's the default value

File: account_payment_partner/models/res_partner.py
Comment (astirpe) [inline account_payment_partner/models/res_partner.py]: string="Customer Payment Mode", could be removed here, as it's the default value

File: account_payment_partner/readme/CONFIGURE.rst
Comment (astirpe) [inline account_payment_partner/readme/CONFIGURE.rst]: You an remove file `CONFIGURE.rst`, it's useless

**Review by astirpe — CHANGES_REQUESTED** [review]
Only code review, not tested

File: account_payment_partner/models/account_move.py
Comment (astirpe) [inline account_payment_partner/models/account_move.py:53]: Since you added these two methods, it would be nice if you can write some tests for the code coverage.

File: account_payment_partner/models/account_move.py
Comment (astirpe) [inline account_payment_partner/models/account_move.py]: If you look at the code of the previous version, this comment is meant to be for the bank assignation, so it should be moved into `def _compute_invoice_partner_bank`

File: account_payment_partner/models/account_move.py
Comment (astirpe) [inline account_payment_partner/models/account_move.py]: `payment_mode_id` is already passed in the depends(...), so this should be:

`pay_mode = move.payment_mode_id`

File: account_payment_partner/models/account_move.py
Comment (astirpe) [inline account_payment_partner/models/account_move.py]: The last three lines should be fixed, to prevent a singleton error:

```python
                    move.invoice_partner_bank_id = False
                elif not move.payment_mode_id:
                    move.invoice_partner_bank_id = False
```

Moreover, according to the previous version of the module, if `self.type` is not equal to `'in_invoice'`, the bank assignation should not be done. So this entire block of code could be replaced with `self.invoice_partner_bank_id = False`.

**Review by astirpe — CHANGES_REQUESTED** [review]

File: account_payment_partner/models/account_move.py
Comment (dynapps) [reply account_payment_partner/models/account_move.py]: supplier_payment_mode_id != payment_mode_id

File: account_payment_partner/models/account_move.py
Comment (dynapps) [reply account_payment_partner/models/account_move.py]: Done!

Comment (rven) [reply account_payment_partner/models/account_move.py:53]: Done!

File: account_payment_partner/models/account_move.py
Comment (rven) [reply account_payment_partner/models/account_move.py]: Done!

File: account_payment_partner/models/account_move.py
Comment (rven) [reply account_payment_partner/models/account_move.py]: Done!

File: account_payment_partner/models/account_move.py
Comment (rven) [reply account_payment_partner/models/account_move.py]: Comment is moved and a little bit adjusted

**Review by ThomasBinsfeld — APPROVED** [review]

**Review by astirpe — APPROVED** [review]

**Comment by pedrobaeza** [comment]
There's a report in Odoo with a possible problem of this module:

https://github.com/odoo/odoo/issues/42149

**Comment by ghost** [comment]
V13 Js error on Accounting -> Journal Items

When i try open journal item from **Accounting -> Journal Items** view show .js error.
after activating debug mode with assets, console show error **"Error: Unknown field state in domain**"
in /web/static/src/js/core/domains.js line 68

error reason is ccount_payment_partner\models\account_move_line.py

`payment_mode_id = fields.Many2one(
        comodel_name="account.payment.mode",
        compute="_compute_payment_mode",
        store=True,
        ondelete="restrict",
        index=True,
        states={"draft": [("readonly", False)]},
    )`

i comment states and error fixed.

**Comment by rven** [comment]
@pedrobaeza @bumblebeey Change is done!

**Comment by SolidGlenn** [comment]
Hi, 
Having trouble on the 13.0 branch.
following error:
odoo.tools.convert.ParseError: "Element '<xpath expr="//p[@t-if='o.invoice_payment_term_id']">' cannot be located in parent view

Error context:
View `report_invoice_payment_mode`
[view_id: 2627, xml_id: n/a, model: n/a, parent_id: 901]" while parsing /opt/odoo/oca-addons/bank-payment/account_payment_partner/views/report_invoice.xml:2, near
<odoo>
    </odoo>


Which is strange, since it actually exists in the report?

**Pull request merged**
